test/docker/aws/aws_test.go

```diff
@@ -0,0 +1,180 @@
+// +build integration
+
+package aws
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/aws/aws-sdk-go/service/sns/snsiface"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+	patronDocker "github.com/beatlabs/patron/test/docker"
+	"github.com/ory/dockertest"
+)
+
+const (
+	testSnsRegion    = "eu-west-1"
+	testSNSTopic     = "test-topic"
+	testSQSQueueName = "test-publish-message"
+)
+
+var (
+	runtime *awsRuntime
+)
+
+func TestMain(m *testing.M) {
+	var err error
+	runtime, err = create(60 * time.Second)
+	if err != nil {
+		fmt.Printf("could not create AWS runtime: %v\n", err)
+		os.Exit(1)
+	}
+
+	exitCode := m.Run()
+
+	ee := runtime.Teardown()
+	if len(ee) > 0 {
+		for _, err := range ee {
+			fmt.Printf("could not tear down containers: %v\n", err)
+		}
+	}
+
+	os.Exit(exitCode)
+}
+
+type awsRuntime struct {
+	patronDocker.Runtime
+}
+
+func create(expiration time.Duration) (*awsRuntime, error) {
+	br, err := patronDocker.NewRuntime(expiration)
+	if err != nil {
+		return nil, fmt.Errorf("could not create base runtime: %w", err)
+	}
+
+	runtime := &awsRuntime{Runtime: *br}
+
+	runOptions := &dockertest.RunOptions{
+		Repository: "localstack/localstack",
+		Tag:        "0.11.2",
+		Env: []string{
+			"LOCALSTACK_SERVICES=sqs,sns",
+			"LOCALSTACK_DEBUG=1",
+			"LOCALSTACK_DATA_DIR=/tmp/localstack/data",
+			"AWS_ACCESS_KEY_ID=test",
+			"AWS_SECRET_ACCESS_KEY=test",
+			"AWS_DEFAULT_REGION=eu-west-1",
+		}}
+	_, err = runtime.RunWithOptions(runOptions)
+	if err != nil {
+		return nil, fmt.Errorf("could not start mysql: %w", err)
+	}
+
+	// wait until the container is ready
+	err = runtime.Pool().Retry(func() error {
```

Comment:
would it make sense to keep track of what we could initialise, so that we avoid doing it again and again (assuming that the components come up slowly ?)

Reply:
what do you propose exactly?

Reply:
something like 
`
var sqsStarted bool
var snsStarted bool
err = runtime.Pool().Retry(func ... )
`
and set the bools, so that we dont do the calls ?
It s minor ... just wondering if it makes sense to you.

Reply:
Makes sense. Done